src/base/ceramicApiKit.ts

```diff
@@ -0,0 +1,197 @@
+                      proposer: "${senderAddress}"
+                      operation: 0
+                      safeTxHash: "${safeTxHash}"
+                      signatures: "${senderSignature}"
+                      submissionDate: "${(new Date()).toISOString()}"
+                      transactionHash: "${safeTxHash}"
+                      confirmationsRequired: ${threshold}
+                  }
+                  clientMutationId: null
+              }
+          ) {
+              clientMutationId
+          }
+      }
+    `);
+
+
+    return Promise.resolve();
+  }
+
+  async getTransaction(safeTxHash: string): Promise<SafeMultisigTransactionResponse> {
+
+    try {
+      if (!this.composeClient) {
+        await this.connect();
+      }
+      const transactionIndex = await this.composeClient.executeQuery(`
+          query TransactionIndex {
+              transactionIndex(
+                  first: 1
+                  filters: { where: { safeTxHash: { equalTo: "${safeTxHash}" } } }
+              ) {
+                  edges {
+                      cursor
+                      node {
+                          id
+                          to
+                          data
+                          safe
+                          nonce
+                          value
+                          proposer
+                          operation
+                          safeTxHash
+                          signatures
+                          submissionDate
+                          transactionHash
+                          confirmationsRequired
+                      }
+                  }
+              }
+          }
+        `);
+
+      const confirmationsIndex: ConfirmationIndexData = await this.composeClient.executeQuery(`
+            query ConfirmationIndex {
+                confirmationIndex(
+                    first: 10
+                    filters: { where: { transactionHash: { equalTo: "${safeTxHash}" } } }
+                ) {
+                    edges {
+                        node {
+                            owner
+                            id
+                            signature
+                            signatureType
+                            submissionDate
+                            transactionHash
+                            confirmationType
+                        }
+                    }
+                }
+            }
+        `) as ConfirmationIndexData;
+      const confirmations = confirmationsIndex.data.confirmationIndex.edges.map((edge) => edge.node).filter((confirmation) => {
+        const { signature } = confirmation;
+        const r = signature.slice(0, 66);
+        const s = `0x${signature.slice(66, 130)}`;
+        let v = parseInt(signature.slice(130, 132), 16);
+        if (v !== 27 && v !== 28) {
+          v = 27;
+        }
+        // normalize signature
+        const normalizedSignature = r + s.slice(2) + (v).toString(16).padStart(2, '0');
```

Comment:
Why is it necessary to separate and then merge this signature?

Reply:
If you're dealing with v fields, ethers.js handles them accordingly.
https://github.com/ethers-io/ethers.js/blob/main/lib.esm/crypto/signature.js#L204

Reply:
Our test case is 31, is it possible that for other chains the value is 30 or some other number?

Reply:
I guess only way to find out is test， would let you know once I done

Reply:
and it does change, 
31 on `base-sepolia`
32 on `sepolia`

Reply:
working on find out more
these are only testnets supported by Safe Genosis